openmc/source.py

```diff
@@ -198,6 +198,43 @@ def to_xml_element(self):
             element.append(self.time.to_xml_element('time'))
         return element
 
+    def plot_energy(self, n_samples=10000, seed=None, axes=None):
+        """Produce a histogram plot of source particle energy.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of sampled values to generate
+        seed : int or None
+            Initial random number seed.
+        axes : matplotlib.axes.Axes, optional
+            Axes for plot
+
+        Returns
+        -------
+        axes : matplotlib.axes.Axes
+            Axes for plot
+
+        """
+
+        energy = self.energy.sample(n_samples=n_samples, seed=seed)
+
+        import matplotlib.pyplot as plt
+
+        # Setup axes is one wasn't passed
+        if axes is None:
+            fig = plt.figure()
+            ax = plt.axes()
```

Comment:
This will create a new figure by default. How about using `plt.gca()` (for Get Current Axis)?

This way, users will be able to effortlessly include this function into their plotting workflow
```suggestion
            ax = plt.gca()
```

Reply:
The current way of combining this function with another plot would be to do:

```python
import matplotlib.pyplot as plt
plt.plot([1, 2, 3])

my_source.plot_energy(ax=plt.gca())

plt.ylabel("My custom y label")

plt.show()
```

Which is imho less convenient than what's proposed in this comment:

```python
import matplotlib.pyplot as plt
plt.plot([1, 2, 3])

my_source.plot_energy()

plt.ylabel("My custom y label")

plt.show()
```

Furthermore calling it without anything done before will automatically create a new figure just as the current implementation.

Reply:
Are you sure that we can remove the ``` fig = plt.figure() ``` line as it is also here https://github.com/openmc-dev/openmc/blob/6549067c43bb00750fe4f3d772795217575380ea/openmc/tracks.py#L166

Reply:
a new figure is created if there's no active axis

Reply:
On this snipset, adding `plt.figure()` or not doesn't change anything
```python
import matplotlib.pyplot as plt
import numpy as np

plt.figure()

n, bins, patches = plt.hist(np.random.random(100))

for i, p in enumerate(patches):
    p.set_alpha((i/len(patches)))


plt.show()
```